tests/filecheck/transforms/convert-stencil-to-ll-mlir.mlir

```diff
@@ -349,5 +349,99 @@ func.func @neg_bounds(%in : !stencil.field<[-32,32]xf64>, %out : !stencil.field<
+// CHECK-NEXT:    %134 = "memref.subview"(%133) {"static_offsets" = array<i64: 4>, "static_sizes" = array<i64: 64>, "static_strides" = array<i64: 1>, "operand_segment_sizes" = array<i32: 1, 0, 0, 0>} : (memref<72xf64>) -> memref<64xf64, strided<[1], offset: 4>>
+// CHECK-NEXT:    %135 = "memref.subview"(%132) {"static_offsets" = array<i64: 4>, "static_sizes" = array<i64: 64>, "static_strides" = array<i64: 1>, "operand_segment_sizes" = array<i32: 1, 0, 0, 0>} : (memref<72xf64>) -> memref<64xf64, strided<[1], offset: 4>>
+// CHECK-NEXT:    %136 = "memref.alloc"() {"operand_segment_sizes" = array<i32: 0, 0>} : () -> memref<64xf64>
+// CHECK-NEXT:    %137 = "memref.subview"(%136) {"static_offsets" = array<i64: -1>, "static_sizes" = array<i64: 64>, "static_strides" = array<i64: 1>, "operand_segment_sizes" = array<i32: 1, 0, 0, 0>} : (memref<64xf64>) -> memref<64xf64, strided<[1], offset: -1>>
+// CHECK-NEXT:    %138 = "arith.constant"() {"value" = 1 : index} : () -> index
+// CHECK-NEXT:    %139 = "arith.constant"() {"value" = 1 : index} : () -> index
+// CHECK-NEXT:    %140 = "arith.constant"() {"value" = 65 : index} : () -> index
+// CHECK-NEXT:    "scf.parallel"(%138, %140, %139) ({
+// CHECK-NEXT:    ^16(%141 : index):
+// CHECK-NEXT:      %142 = "arith.constant"() {"value" = -1 : index} : () -> index
+// CHECK-NEXT:      %143 = arith.addi %141, %142 : index
+// CHECK-NEXT:      %144 = "memref.load"(%135, %143) : (memref<64xf64, strided<[1], offset: 4>>, index) -> f64
+// CHECK-NEXT:      "memref.store"(%144, %137, %141) : (f64, memref<64xf64, strided<[1], offset: -1>>, index) -> ()
+// CHECK-NEXT:      "scf.yield"() : () -> ()
+// CHECK-NEXT:    }) {"operand_segment_sizes" = array<i32: 1, 1, 1, 0>} : (index, index, index) -> ()
+// CHECK-NEXT:    %145 = "arith.constant"() {"value" = 0 : index} : () -> index
+// CHECK-NEXT:    %146 = "arith.constant"() {"value" = 1 : index} : () -> index
+// CHECK-NEXT:    %147 = "arith.constant"() {"value" = 64 : index} : () -> index
+// CHECK-NEXT:    "scf.parallel"(%145, %147, %146) ({
+// CHECK-NEXT:    ^17(%148 : index):
+// CHECK-NEXT:      %149 = "arith.constant"() {"value" = 1 : index} : () -> index
+// CHECK-NEXT:      %150 = arith.addi %148, %149 : index
+// CHECK-NEXT:      %151 = "memref.load"(%137, %150) : (memref<64xf64, strided<[1], offset: -1>>, index) -> f64
+// CHECK-NEXT:      "memref.store"(%151, %134, %148) : (f64, memref<64xf64, strided<[1], offset: 4>>, index) -> ()
+// CHECK-NEXT:      "scf.yield"() : () -> ()
+// CHECK-NEXT:    }) {"operand_segment_sizes" = array<i32: 1, 1, 1, 0>} : (index, index, index) -> ()
+// CHECK-NEXT:    "memref.dealloc"(%137) : (memref<64xf64, strided<[1], offset: -1>>) -> ()
+// CHECK-NEXT:    "func.return"() : () -> ()
+// CHECK-NEXT:  }
+
+  func.func @stencil_two_stores(%59 : !stencil.field<[-4,68]xf64>, %60 : !stencil.field<[-4,68]xf64>, %61 : !stencil.field<[-4,68]xf64>) {
```

Comment:
What do you wish to test with this?

Reply:
So, temps output can be either buffered or stored, because if it is stored already, there's no need to allocate a whole new buffer to read it (the purpose of stencil.buffer).
This tests that this case of chained applys is lowered correctly too! It actually needed a tweak in the lowering indeed, visibile in the corresponding commit.